lldb/docs/resources/lldbdap.md

```diff
@@ -1,195 +1,147 @@
-# command.
-npm run package
-npm run vscode-install
-```
-
-### Deploying for Visual Studio Code
+Welcome to the `llb-dap` documentation!
 
-The easiest way to deploy the extension for execution on other machines requires
-copying `lldb-dap` and its dependencies into a`./bin` subfolder and then create a
-standalone VSIX package.
+`lldb-dap` brings the power of `lldb` into any editor or IDE that supports the
+[Debug Adapter Protocol (DAP)](https://microsoft.github.io/debug-adapter-protocol/).
 
-```bash
-cd /path/to/lldb/tools/lldb-dap
-mkdir -p ./bin
-cp /path/to/a/built/lldb-dap ./bin/
-cp /path/to/a/built/liblldb.so ./bin/
-npm run package
-```
+## Prerequisites
 
-This will produce the file `./out/lldb-dap.vsix` that can be distributed. In
-this type of installation, users don't need to manually set the path to
-`lldb-dap`. The extension will automatically look for it in the `./bin`
-subfolder.
+In order to begin debugging with `lldb-dap`, you may first need to acquire the
+`lldb-dap` binary from an LLVM distribution. For general LLVM releases visit
+https://releases.llvm.org/ or check your systems preferred package manager for
+the `lldb` package.
 
-*Note: It's not possible to use symlinks to `lldb-dap`, as the packaging tool
-forcefully performs a deep copy of all symlinks.*
+In some cases, a language specific build of `lldb` / `lldb-dap` may also be
+available as part of the languages toolchain. For example the
+[swift language](https://www.swift.org/) toolchain includes additional language
+integrations in `lldb` and the toolchain builds provider both the `lldb` driver
+binary and `lldb-dap` binary.
 
-*Note: It's possible to use this kind flow for local installations, but it's
-not recommended because updating `lldb-dap` requires rebuilding the extension.*
+## IDE Integration
 
-## Formatting the Typescript code
+In addition to the `lldb-dap` binary, some IDEs have additional extensions to
+support debugging.
 
-This is also very simple, just run:
+- Visual Studio Code -
+[LLDB DAP Extension](https://marketplace.visualstudio.com/items?itemName=llvm-vs-code-extensions.lldb-dap)
+<!-- Add other IDE integrations. -->
 
-```bash
-npm run format
-```
+## Launching a program
 
-## Working with the VS Code extension from another extension
+To launch an executable for debugging, first define a launch configuration tells
+`lldb-dap` how to launch the binary.
 
-The VS Code extension exposes the following [VS Code
-commands](https://code.visualstudio.com/api/extension-guides/command),
-which can be invoked by other debugger extensions to leverage this extension's
-settings and logic. The commands help resolve configuration, create adapter
-descriptor, and get the lldb-dap process for state tracking, additional
-interaction, and telemetry.
+A simple launch configuration may look like
 
+```json
+{
+  "type": "lldb-dap",
+  "request": "launch",
+  "name": "Debug a.out",
+  "program": "a.out"
+}
 ```
```

Comment:
Aren't attach & launch configurations a VS Code concept? Presumably we need to mention them here to motivate all the configuration options we support. Maybe we should frame it as "depending on your editor, you have to specify how to launch or attach". For VS Code, bla bla bla.